pepsico/isimiptozarr.py

```diff
@@ -0,0 +1,284 @@
+        Either a string glob in the form "path/to/my/files/*.nc"
+        or an explicit list of files to open. Paths can be given as strings
+        or as pathlib Paths.
+    var_name : str
+        name of the ENACTS variable in the nc files
+    time_res : str, optional
+        indicates the time resolution of the set of files.
+        Default is "daily" and other option is "dekadal"
+    zarr_resolution : real, optional
+        spatial resolution to regrid to.
+    chunks : int, tuple of int, "auto" or mapping of hashable to int, optional
+        Chunk sizes along each dimension X, Y and T.
+    
+    Returns
+    -------
+    Xarray.Dataset containing variable `var_name` and coordinates X, Y and T
+
+    See Also
+    --------
+    xarray.open_mfdataset, set_up_dims, regridding, xarray.DataArray.chunk
+    """
+    data = xr.open_mfdataset(
+        paths,
+        preprocess=partial(
+            set_up_dims,
+            time_res=time_res,
+            time_dim=time_dim,
+            lon_dim=lon_dim,
+            lat_dim=lat_dim
+        ),
+        parallel=False,
+    )[var_name]
+    print(data)
+    if zarr_resolution != None:
+        print("attempting regrid")
+        data = regridding(data, zarr_resolution)
+    return xr.Dataset().merge(data.chunk(chunks=chunks))
+
+
+def convert(
+    input_path,
+    output_path,
+    var_name,
+    time_res="daily",
+    zarr_resolution=None,
+    chunks={},
+    file_var_pattern="*.nc",
+    time_dim=None,
+    lon_dim="Lon",
+    lat_dim="Lat",
+):
+    """Converts a set of ENACTS files into zarr store.
+
+    Either create a new one or append an existing one
+
+    Parameters
+    ----------
+    input_path : str
+        path where the ENACTS nc files are
+    output_path : str
+        path where the zarr store is (to append to) or will be (to create).
+        To create, (last element of the) path is expected not to exist.
+        To append, path is expected to point to a zarr store.
+    var_name : str
+        name of the ENACTS variable in the nc files
+    time_res : str, optional
+        indicates the time resolution of the set of files.
+        Default is "daily" and other option is "dekadal"
+    zarr_resolution : real, optional
+        spatial resolution to regrid to.
+    chunks : int, tuple of int, "auto" or mapping of hashable to int, optional
+        Chunk sizes along each dimension X, Y and T.
+        
+    Returns
+    -------
+    output_path : where the zarr store has been written
+
+    See Also
+    --------
+    calc.read_zarr_data, filename2datetime64, nc2xr, xarray.Dataset.to_zarr
+    """
+    print(f"converting files for: {time_res} {var_name}")
+    netcdf = list(sorted(Path(input_path).glob(file_var_pattern)))
+    if Path(output_path).is_dir() :
+        current_zarr = calc.read_zarr_data(output_path)
+        last_T_zarr = current_zarr["T"][-1]
+        last_T_nc = filename2datetime64(netcdf[-1], time_res=time_res)
```

Comment:
This assumes that the file that comes last in alphabetical order is the latest one chronologically. That might not always be a valid assumption, e.g. I've seen filenames that used month names instead of numbers (Aug comes before Jan alphabetically, but not chronologically) and ones that put the year last (01-2024 comes before 06-2023 alphabetically but not chronologically).

It's ok for a dataset-specific script where we know that the filename pattern supports it, but we need to be careful of this when factoring out a reusable library.

Reply:
Right. They definitely are filenames that of which alphanumerical order is not chronological order. Shall I file this in an issue? Neither ENACTS nor ISIMIP are that case.